### ISSUE TITLE
Unable to Write Rules to Check Properties

### ISSUE THREAD

**Issue by andrei572** [stoplightio/spectral#1131]
Is there possibly a (known) issue with jsonpath in spectral rules? I am trying to create a custom rule that would check all properties in a spec for a description present using the `truthy` function, except for the ones that are references, and I am unable to successfully iterate over $..properties.* and check for the presence of $ref. Spectral seems to be selecting OAS schemas that don't even have the `properties` key.

This seems like like it should work to find all properties that are not references: $..properties.*[?(!@['$ref'])].

But it doesn't.

**Comment by P0lip** [comment]
Hey!
You need to set `resolved` to false on your rule.
Otherwise it operates on a resolved document that has no $refs left.

**Comment by andrei572** [comment]
Thanks. I ma getting different results now, but iterating over properties still does not work and it's not finding problems in the schema defs when I deliberately introduce them in properties, and bringing back false positives that don't make sense.

**Comment by andrei572** [comment]
Okay, this seems to be working:

```
rules:
  property-examples-test:
    description: 'Object properties in schema definitions must have descriptions.'
    given: "$..['properties'].[?(!@.$ref && @property !== 'properties')]"
    severity: error
    resolved: false
    then:
      field: description
      function: 'truthy'
```

Thanks again!

**Comment by P0lip** [comment]
Great, glad to hear you were able to make it working.
I'll close the issue. Shall you have any further concerns, feel free to raise them here. :)
